### ISSUE TITLE
Support back-slash in load directive path

### ISSUE THREAD

**Issue by aateeque** [cake-build/cake#1417]
### What You Are Seeing?
Backslashes in `#load` directive in Cake scripts are URI escaped to %5C which results in 
```
Could not find script 'C:/dev/Tools/TeamCity/buildAgent/work/de8c636c0595ba4a/.%5CTools%5Cfluid.DefaultBuild%5Cgitversion.cake'.
```

### What is Expected?
The script should execute

### What version of Cake are you using?
Latest on develop branch, sha: 62d3aea

### Are you running on a 32 or 64 bit system?
64-bit

### What environment are you running on?  Windows? Linux? Mac?
Windows

### Are you running on a CI Server?  If so, which one?
No; running in Visual Studio 2015 (though the path above points to TC, its just the checkout dir)

### How Did You Get This To Happen? (Steps to Reproduce)
Believe changes in #1359 have resulted in this. 

### Output Log
```
Error: Cake.Core.CakeException: Could not find script 'C:/dev/Tools/TeamCity/buildAgent/work/de8c636c0595ba4a/.%5CTools%5Cfluid.DefaultBuild%5Cgitversion.cake'.
   at Cake.Core.Scripting.Analysis.ScriptAnalyzerContext.Analyze(FilePath path) in C:\dev\code\cake\cake\src\Cake.Core\Scripting\Analysis\ScriptAnalyzerContext.cs:line 66
   at Cake.Core.Scripting.Processors.Loading.FileLoadDirectiveProvider.Load(IScriptAnalyzerContext context, LoadReference reference) in C:\dev\code\cake\cake\src\Cake.Core\Scripting\Processors\Loading
\FileLoadDirectiveProvider.cs:line 44
   at Cake.Core.Scripting.Processors.LoadDirectiveProcessor.AddToContext(IScriptAnalyzerContext context, Uri uri) in C:\dev\code\cake\cake\src\Cake.Core\Scripting\Processors\LoadDirectiveProcessor.cs:
line 47
   at Cake.Core.Scripting.Processors.UriDirectiveProcessor.Process(IScriptAnalyzerContext context, String line, String& replacement) in C:\dev\code\cake\cake\src\Cake.Core\Scripting\Processors\UriDire
ctiveProcessor.cs:line 41
   at Cake.Core.Scripting.Analysis.ScriptAnalyzer.<>c__DisplayClass6_2.<AnalyzeCallback>b__0(LineProcessor p) in C:\dev\code\cake\cake\src\Cake.Core\Scripting\Analysis\ScriptAnalyzer.cs:line 117
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
   at Cake.Core.Scripting.Analysis.ScriptAnalyzer.AnalyzeCallback(IScriptAnalyzerContext context) in C:\dev\code\cake\cake\src\Cake.Core\Scripting\Analysis\ScriptAnalyzer.cs:line 117
   at Cake.Core.Scripting.Analysis.ScriptAnalyzerContext.Analyze(FilePath path) in C:\dev\code\cake\cake\src\Cake.Core\Scripting\Analysis\ScriptAnalyzerContext.cs:line 77
   at Cake.Core.Scripting.Analysis.ScriptAnalyzer.Analyze(FilePath path) in C:\dev\code\cake\cake\src\Cake.Core\Scripting\Analysis\ScriptAnalyzer.cs:line 90
   at Cake.Core.Scripting.ScriptRunner.Run(IScriptHost host, FilePath scriptPath, IDictionary`2 arguments) in C:\dev\code\cake\cake\src\Cake.Core\Scripting\ScriptRunner.cs:line 133
   at Cake.Commands.BuildCommand.Execute(CakeOptions options) in C:\dev\code\cake\cake\src\Cake\Commands\BuildCommand.cs:line 34
   at Cake.CakeApplication.Run(CakeOptions options) in C:\dev\code\cake\cake\src\Cake\CakeApplication.cs:line 46
   at Cake.Program.Main() in C:\dev\code\cake\cake\src\Cake\Program.cs:line 75
```

**Comment by aateeque** [comment]
This is a breaking change in the latest bits; script works fine with 0.17.0

**Comment by aateeque** [comment]
Think the problem is in `UriDirectiveProcessor.ParseUriFromTokens`; we could swap back slashes for forward slashes here before parsing into `Uri`. @patriksvensson thoughts please?

**Comment by patriksvensson** [comment]
Back slashes are not valid in URI:s :(

**Comment by aateeque** [comment]
Should have included this above, my cake script has:
```
#l .\Tools\fluid.DefaultBuild\nuget.cake
```

which I think is valide path to a cake script to load, right?

**Comment by augustoproiete** [comment]
Closing as resolved in milestone/release v0.30.0

```csharp
#l .\Tools\fluid.DefaultBuild\nuget.cake
```

---

```
Looking for modules...
No modules found to install.
Module directory does not exist.
Analyzing build script...
Analyzing C:/cake/tool/1.0.0/test-load-backslash.cake...
Analyzing C:/cake/tool/1.0.0/Tools/fluid.DefaultBuild/nuget.cake...
Processing build script...
```
